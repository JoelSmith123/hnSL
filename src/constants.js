const dataBaseKey =  '327282-findstuf-NRPF8M1F';

export {
  dataBaseKey
}