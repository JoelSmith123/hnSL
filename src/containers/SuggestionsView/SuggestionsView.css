.SuggestionsView {
  background-color: #fff;
  height: calc(100% - 50px);
  margin-left: 350px;
  padding: 50px;
}

.suggestion-cards-container {
  display: grid;
  grid-gap: 30px;
  grid-template-columns: 1fr 1fr;
  margin-top: 50px;
}

.suggestions-view-title {
  margin: 0;
}