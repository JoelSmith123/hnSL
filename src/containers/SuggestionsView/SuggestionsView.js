import React, { Component } from 'react'
import { connect } from 'react-redux'
import uuid from 'uuid'
import { dataBaseKey }  from '../../../src/constants.js'
import { fetchSuggestions } from '../../thunks/fetchSuggestions.js'
import SuggestionCard from '../SuggestionCard/SuggestionCard.js'
import './SuggestionsView.css'

export class SuggestionsView extends Component {

  async componentDidMount() {
    let searchRequest = Object.values(this.props.filters.filterInputs).map(query => {
      query = query.split(/[ ]+/).join('+')
      return query
    }).join('+&2c+')
    const url = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=${searchRequest}&k=${dataBaseKey}&info=1`
    await this.props.fetchSuggestions(url)
  }

  render() {
    return (
      <div className='SuggestionsView'>
        <div className='suggestion-cards-container'>
          {
            this.props.suggestions ? 
              this.props.suggestions.Similar.Results.map(suggestion => {
                return <SuggestionCard {...suggestion} key={uuid()} />
              })
            : null
          }
        </div>
      </div>
    )
  }
}

export const mapStateToProps = (state) => {
  return {
    suggestions: state.suggestions.suggestions,
    filters: state.filters
  }
}
  
export const mapDispatchToProps = (dispatch) => ({
  fetchSuggestions: (url) => dispatch(fetchSuggestions(url))
})

export default connect(mapStateToProps, mapDispatchToProps)(SuggestionsView)
