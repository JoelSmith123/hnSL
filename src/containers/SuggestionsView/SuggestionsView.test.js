import React from 'react'
import { shallow } from 'enzyme'
import { dataBaseKey }  from '../../../src/constants.js'
import { SuggestionsView, mapStateToProps, mapDispatchToProps } from './SuggestionsView.js'
jest.mock('../../thunks/fetchSuggestions.js')

describe('SuggestionsView', () => {
  let mockFetch
  let mockFilters
  let wrapper

  beforeEach(() => {
    mockFetch = jest.fn()
    mockFilters = {'input-0': 'red hot chili peppers', 'input-1': 'pulp fiction'}
    wrapper = shallow(<SuggestionsView suggestions={[{movie: 'movie-one'}]} 
                                       fetchSuggestions={ mockFetch } 
                                       filters={ mockFilters } 
                                       category={'movies'}
                      />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call fetchSuggestions on componentDidMount', async () => {
    const searchRequest = 'red+hot+chili+peppers+&2c+pulp+fiction'
    const category = 'movies'
    const url = `https://cors-anywhere.herokuapp.com/https://tastedive.com/api/similar?q=${searchRequest}&k=${dataBaseKey}&info=1&type=${category}`

    await wrapper.instance().componentDidMount()

    expect(mockFetch).toHaveBeenCalledWith(url)
  })
})

describe('mapStateToProps', () => {
  it('should return an object with expected values', () => {
    const mockState = {
      suggestions: [{movie: 'movie-one'}],
      filters: {filterOne: 'filter'},
      category: 'games',
      goodLuck: 'plenty'
    }
    const expected = {
      suggestions: [{movie: 'movie-one'}],
      filters: {filterOne: 'filter'},
      category: 'games'
    }

    const mappedProps = mapStateToProps(mockState)

    expect(mappedProps).toEqual(expected)
  })
})

describe.skip('mapDispatchToProps', () => {
  it.skip('calls dispatch with a fetchSuggestions action when fetchSuggestions is called', () => {
    const mockDispatch = jest.fn()
    const mockUrl = 'www.google.com'
    const actionToDispatch = fetchSuggestions(mockUrl)

    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.fetchSuggestions(mockUrl)

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })
})