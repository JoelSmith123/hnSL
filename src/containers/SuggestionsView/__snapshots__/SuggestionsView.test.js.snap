// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuggestionsView should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SuggestionsView
    fetchSuggestions={[MockFunction]}
    suggestions={
      Array [
        Object {
          "movie": "movie-one",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="suggestion-cards-container"
      />,
      "className": "SuggestionsView",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": null,
        "className": "suggestion-cards-container",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="suggestion-cards-container"
        />,
        "className": "SuggestionsView",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": null,
          "className": "suggestion-cards-container",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
